paparazzi-agent/src/main/java/app/cash/paparazzi/agent/AgentTestRule.kt

```diff
@@ -0,0 +1,23 @@
+package app.cash.paparazzi.agent
+
+import net.bytebuddy.agent.ByteBuddyAgent
+import org.junit.rules.TestRule
+import org.junit.runner.Description
+import org.junit.runners.model.Statement
+
+class AgentTestRule : TestRule {
+  override fun apply(
+    base: Statement,
+    description: Description
+  ) = object : Statement() {
+    override fun evaluate() {
+      ByteBuddyAgent.install()
+      InterceptorRegistrar.registerMethodInterceptors()
+
+      try {
+        base.evaluate()
+      } finally {
+      }
```

Comment:
Remove

Reply:
Not sure if you mean only the `finally` or the whole try/finally?

Someone could wrap this in a RuleChain and a preceding rule could throw, so I figure keeping both would make sense?

Reply:
Merging, will follow up on this.

Reply:
The code as-is does nothing. The whole try/finally can be removed.

Reply:
removing the try-catch would result in never running the test….
```
class MyTest {
    @get:Rule
    val paparazzi =Paparazzi()  // magically adds AgentRule before itself

    @Test fun sometest() {  throw IllegalStateException("test ran!") }
}
```

with `base.evaluate`:
paparazzi.evaluate -> agentrule.evaluate -> invokemethod.evaluate which runs the test method (boom)

without `base.evaluate`:
paparazzi.evaluate -> agentrule.evaluate -> never runs test.  False success.

Reply:
Keep `base.evaluate()`. Just remove the try/finally. It catches no exceptions and runs no code on an exception being thrown.

Reply:
https://github.com/cashapp/paparazzi/pull/133